### ISSUE TITLE
[Usage]: How to use RLinf to sft pi05 on CALVIN dataset

### ISSUE THREAD

**Issue by 1768637871** [RLinf/RLinf#762]
### Problem description

Thanks for your great job! Does RLinf support finetuning pi05 model on CALVIN Benchmark? I use the mirror rlinf/rlinf:agentic-rlinf0.1-calvin to try this but fail. The InternRobotics/InternData-Calvin_ABC dataset seems to be not compatible wich RLinf. Besides, i encounter a series of problems with fsdp. Can you provide a detailed process on sft pi05 on CALVIN, thanks a lot!
the config are show as follow:
`
defaults:
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: 0-0

runner:
  task_type: sft
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "test_openpi_calvin"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1
  val_check_interval: -1
  save_interval: 10

data:
  data_path: "/apdcephfs_sh2/share_1044881/xx/huggingface/hub/"

algorithm:
  adv_type: gae

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 1
  global_batch_size: 16
  seed: 0

 
  model:
    precision: null
    model_path: "/apdcephfs_sh2/share_1044881/xx/huggingface/hub/pi05_base_pytorch"
    num_action_chunks: 4 # interface for the env
    add_value_head: True
    openpi:
      config_name: "pi05_calvin"
      detach_critic_input: True

  optim:
    lr: 7.91e-6
    value_lr: 1.55e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-08
    weight_decay: 0.01
    clip_grad: 1.0

  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "no_shard"
    use_orig_params: False
    gradient_checkpointing: False # for openpi, gradient checkpointing is not supported, please do not change this value
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

reward:
  use_reward_model: False

critic:
  use_critic_model: False
`

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by chenkang455** [comment]
@1768637871 Sry for the late reply. For Calvin SFT, we conduct it on the vallina openpi repo. It should be easy by copying our calvin_dataconfig, calvin_policy and config files to the official openpi repo and reproducing the SFT training.

**Comment by 1768637871** [comment]
@chenkang455 Thanks for your reply! I will try this.

**Comment by 1768637871** [comment]
@chenkang455  Hello, I also have a question about the evaluation of the CALVIN benchmark. I used the command listed below, but I did not obtain the success rate for each task in CALVIN, nor the average execution length shown in the figure. I was wondering whether there might be an issue with my configuration, or if these metrics need to be implemented manually.

<img width="1414" height="492" alt="Image" src="https://github.com/user-attachments/assets/9c2b1ad7-456a-4e45-be91-141e6a41f259" />

I use the command `bash examples/embodiment/eval_embodiment.sh calvin_d_d_eval_openpi_pi05` and the config calvin_d_d_eval_openpi_pi05.yaml is shown as below.


```
defaults:
  - env/calvin_d@env.train
  - env/calvin_d@env.eval
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: 0

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "calvin_d_d_eval_openpi_pi05"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1
  max_steps: -1

  only_eval: True
  val_check_interval: -1
  save_interval: 100

  resume_dir: null # Optional: path to a saved checkpoint directory, such as 'checkpoints/global_step_10'. If not None, it will be used to resume training.
  ckpt_path: null  # Optional: path to a .pt checkpoint. If not None, it will be loaded after the model is instantiated (for evaluation).

algorithm:
  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 1
  reward_coef: 1.0

  rollout_epoch: 1
  eval_rollout_epoch: 1 # set eval_rollout_epoch > 0 when enable runner.only_eval or runner.val_check_interval > 0
  reward_type: chunk_level
  logprob_type: chunk_level
  entropy_type: token_level

  update_epoch: 1
  adv_type: gae
  loss_type: actor_critic
  loss_agg_func: "token-mean" 
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  filter_rewards: False
  rewards_lower_bound: 0.1
  rewards_upper_bound: 0.9
  # params for generation
  sampling_params:
    do_sample: True
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 50
    top_p: 1.0
    repetition_penalty: 1.0
    add_BOS: False

  # length argument for autoregressive sampling
  # max length means max amount of tokens to generate
  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"
  enable_offload: False

  train:
    total_num_envs: 64
    max_episode_steps: 480 # for truncation
    max_steps_per_rollout_epoch: 480
    group_size: ${algorithm.group_size}
    reward_coef: ${algorithm.reward_coef}
    video_cfg:
      save_video: True
      video_base_dir: ${runner.logger.log_path}/video/train
  eval:
    total_num_envs: 50
    auto_reset: True
    ignore_terminations: True
    max_episode_steps: 480
    max_steps_per_rollout_epoch: 9600
    use_fixed_reset_state_ids: True
    use_ordered_reset_state_ids: True
    is_eval: True
    video_cfg:
      save_video: True
      video_base_dir: ${runner.logger.log_path}/video/eval

rollout:
  group_name: "RolloutGroup"
  generation_backend: "huggingface"
  recompute_logprobs: False
  enable_offload: False
  pipeline_stage_num: 1

  model:
    model_path: "/x/huggingface/hub/RLinf-Pi05-CALVIN-ABC-D-SFT"
    # model_path: "/x/work/RLinf/logs/20260203-16:27:18-calvin_d_d_ppo_openpi_pi05/calvin_d_d_ppo_openpi_pi05/checkpoints/global_step_600/actor/model_state_dict"
    precision: ${actor.model.precision}

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"

  micro_batch_size: 128
  global_batch_size: 128 # maximum : global batch size = micro batch size * actor num_processes.
  seed: 42
  enable_offload: False

  model:
    model_path: "/x/huggingface/hub/RLinf-Pi05-CALVIN-ABC-D-SFT"
    # model_path: "/x/work/RLinf/logs/20260203-16:27:18-calvin_d_d_ppo_openpi_pi05/calvin_d_d_ppo_openpi_pi05/checkpoints/global_step_600/actor/model_state_dict"
    model_type: "openpi"
    num_action_chunks: 5
    add_value_head: True
    # openpi specific parameters
    openpi:
      config_name: "pi05_calvin"
      num_images_in_input: 2
      noise_level: 0.5
      joint_logprob: False
      value_after_vlm: True
      value_vlm_mode: "mean_token"

  optim:
    lr: 5.0e-6
    value_lr: 1.0e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-8
    clip_grad: 1.0
    weight_decay: 0.01
    critic_warmup_steps: 0

  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "no_shard"
    gradient_checkpointing: False # for openpi, gradient checkpointing is not supported, please do not change this value

reward:
  use_reward_model: False

critic:
  use_critic_model: False
```

**Comment by chenkang455** [comment]
@1768637871 Hi, if you want to get the specific length metric, you should try the single script for evaluation mentioned in the doc. 


Check here: https://github.com/RLinf/RLinf/blob/main/toolkits/eval_scripts_openpi/README.md

**Comment by 1768637871** [comment]
@chenkang455 Thanks for your kind reply. It really helps me a lot!
